src/kemal/param_parser.cr

```diff
@@ -57,6 +62,23 @@ module Kemal
       end
     end
 
+    def parse_file_upload
+      HTTP::FormData.parse(@request) do |field, data, meta, headers|
+        if !meta.nil?
+          filename = meta.filename
+          if !filename.nil?
+            tempfile = Tempfile.new(filename)
+            ::File.open(tempfile.path, "w") do |file|
+              IO.copy(data, file)
+            end
+            @files[field] = FileUpload.new(tmpfile: tempfile, tmpfile_path: tempfile.path, meta: meta, headers: headers)
+          else
+            @body[field] = data.gets_to_end
```

Comment:
There isn't really a good size for a formdata body, which is why providing access only to the IO or a tempfile is good. For a urlencoded body I'd say 1MB max, but allow it to be tweaked by the user of course.